### PR TITLE
op: remove LXOR op for floating point

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Floating point logical operations are not in the standard and it does not make sense. Thus remove them.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch3/tcp

**Comment by raffenet** [comment]
Could have sworn we did this already 😄. We just missed LXOR d81fc87980ed717d7afab3095df1f1676369240e`.

**Comment by hzhou** [comment]
> Could have sworn we did this already 😄. We just missed LXOR [d81fc87](https://github.com/pmodels/mpich/commit/d81fc87980ed717d7afab3095df1f1676369240e)`.

I see :)

File: src/mpi/coll/op/oplxor.c [hunk 1]
```diff
@@ -55,10 +55,6 @@ void MPIR_LXOR(void *invec, void *inoutvec, MPI_Aint * Len, MPI_Datatype * type)
                 MPIR_OP_TYPE_GROUP(C_INTEGER_EXTRA)
                 MPIR_OP_TYPE_GROUP(FORTRAN_INTEGER)
                 MPIR_OP_TYPE_GROUP(FORTRAN_INTEGER_EXTRA)
-                /* We previously supported floating point types, although I question
-                 * their utility in logical boolean ops [goodell@ 2009-03-16] */
-                MPIR_OP_TYPE_GROUP(FLOATING_POINT)
-                MPIR_OP_TYPE_GROUP(FLOATING_POINT_EXTRA)
```
Comment (raffenet) [inline src/mpi/coll/op/oplxor.c:61]: Should remove these later in the file in `MPIR_LXOR_check_dtype`? It looks like we left floating pointing types as valid there for all the logical functions.

**Comment by hzhou** [comment]
test:mpich/ch3/tcp

File: src/mpi/coll/op/oplxor.c (hunk 1 above)
Comment (hzhou) [reply src/mpi/coll/op/oplxor.c:61]: fixed.

Comment (raffenet) [reply src/mpi/coll/op/oplxor.c:61]: There were also instances of these in opland.c and oplor.c. I just pushed an extra commit to the branch, which can be squashed.

**Review by raffenet — APPROVED** [review]

**Pull request merged**